writer.go

```diff
@@ -117,78 +101,88 @@ type WriterResponseEmitter struct {
 	emitted bool
 }
 
-func (re *WriterResponseEmitter) SetEncoder(mkEnc func(io.Writer) Encoder) {
+func (re *writerResponseEmitter) SetEncoder(mkEnc func(io.Writer) Encoder) {
 	re.enc = mkEnc(re.w)
 }
 
-func (re *WriterResponseEmitter) SetError(v interface{}, errType cmdkit.ErrorType) {
+func (re *writerResponseEmitter) CloseWithError(err error) error {
+	cwe, ok := re.c.(interface{ CloseWithError(error) error })
+	if ok {
+		return cwe.CloseWithError(err)
+	}
+
+	if err == nil || err == io.EOF {
+		return re.Close()
+	}
+
+	return errors.New("provided closer does not support CloseWithError")
+}
+
+func (re *writerResponseEmitter) SetError(v interface{}, errType cmdkit.ErrorType) {
 	err := re.Emit(&cmdkit.Error{Message: fmt.Sprint(v), Code: errType})
```

Comment:
Shouldn't we remove this function?

Reply:
We can't just remove it, because then it wouldn't satisfy the interface anymore. But I made it simply log an error.